### PR TITLE
Jenkinsfile - Add publish process

### PR DESCRIPTION
**Pull request by nreal**
# Summary
This PR allows publishing in Pre-release and final version to CD

could you review it?

### CODE REVIEW

**Comment by andresmoschini** [comment]
@juanasas maybe you are interested on review this PR

File: build-n-publish.sh [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+imageName=fromdoppler/doppler-currency
+canonicalTag=${preReleasePrefix}${versionFullForTag}
+
+docker build \
+    -t ${imageName}:${canonicalTag} \
+    --build-arg EncryptionSettings__InitVectorAsAsciiString="$EncryptionSettings__InitVectorAsAsciiString" \
+    --build-arg EncryptionSettings__SaltValueAsAsciiString="$EncryptionSettings__SaltValueAsAsciiString" \
+    --build-arg EncryptionSettings__Password="$EncryptionSettings__Password" \
+    --build-arg version=${preReleasePrefix}${versionFull} \
```
Comment (andresmoschini) [inline build-n-publish.sh]: I think that these lines are not required

File: Doppler.Currency/appsettings.json [hunk 2]
```diff
@@ -18,10 +18,10 @@
   "BnaService": {
     "Endpoint": "https://bna.com.ar/Cotizador/HistoricoPrincipales?id=billetes&filtroDolar=1&filtroEuro=0",
     "ValidationHtml": "Dolar U.S.A",
-    "NoQuotation": "There are no pending quotes for that date."
+    "NoCurrency": "There are no pending quotes for that date."
   },
   "SlackHook": {
     "Url": "[SECRET KEY]",
-    "Text": "UsdQuotation Service can't get the USD quotation of the Bank of Nacion Argentina today, Please check Html format"
+    "Text": "Doppler Currency Service can't get the USD quotation of the Bank of Nacion Argentina today, Please check Html format"
```
Comment (andresmoschini) [inline Doppler.Currency/appsettings.json]: These changes are not related with publish, what do you think about moving them to another PR?

File: build-n-publish.sh (hunk 1 above)
Comment (andresmoschini) [reply build-n-publish.sh]: O, the last one is necessary for this:

https://github.com/MakingSense/doppler-forms/blob/master/Dockerfile#L39-L41

So, keep it in this way:

```
docker build \
    -t ${imageName}:${canonicalTag} \
    --build-arg version=${preReleasePrefix}${versionFull} \
    .
```

File: Doppler.Currency/appsettings.json (hunk 2 above)
Comment (nreal) [reply Doppler.Currency/appsettings.json]: Yes, of course, I'll create another PR with more improvement for this repo

File: build-n-publish.sh (hunk 1 above)
Comment (nreal) [reply build-n-publish.sh]: Ok, I'll add line 39 and 40 in Dockerfile file and remove unnecessary lines 102, 103 and 104 in build-n-publish.sh

**Comment by andresmoschini** [comment]
Beautiful!

@juanasas, @CarlosPintos, @cbernat, @elsupergomez, @swidzinskiMS, @fgchaio what do you think? is it ready to be merged?

File: Dockerfile [hunk 3]
```diff
@@ -22,4 +22,6 @@ WORKDIR /app
 EXPOSE 80
 EXPOSE 443
 COPY --from=publish /app/publish .
+ARG version=unknown
+RUN echo $version > /app/wwwroot/version.txt
```
Comment (andresmoschini) [inline Dockerfile]: Oh, it seems that `wwwroot` is not the right place in your project: 

image

Comment (nreal) [reply Dockerfile]: Done, I removed the subfolder `wwwroot` of the path

File: Dockerfile [hunk 4]
```diff
@@ -22,4 +22,6 @@ WORKDIR /app
 EXPOSE 80
 EXPOSE 443
 COPY --from=publish /app/publish .
+ARG version=unknown
+RUN echo $version > /app/version.txt
```
Comment (andresmoschini) [inline Dockerfile]: I think that accessing to `version.txt` is a great thing and very useful.

By the moment your application is not ready, so, what do you think about adding a TODO comment?

```suggestion
# TODO: configure static files in the service and copy version.txt to the right folder
RUN echo $version > /app/version.txt
```

**Comment by cbernat** [comment]
Very nice Naza! 👍
